fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Response.kt

```diff
@@ -12,7 +16,14 @@ class Response {
     var httpContentLength = 0L
 
     //data
-    var data = ByteArray(0)
+    internal var dataStream: InputStream = ByteArrayInputStream(ByteArray(0))
```

Comment:
Ah .. I might have overlooked that usecase! I think your point is valid indeed. I will open a patch on this for v 1.8.1. 😓  really sorry for this!